gameoflife.php

```diff
@@ -20,21 +23,29 @@
  * @var BaseOutput $allOutputs
  */
 $allOutputs = [];
-$maxSteps = 500;
+/**
+ * @var BaseRule $allRules
```

Comment:
The correct docblock would be `@var BaseRule[] $allRules` to denote that this is an array of BaseRule objects.